code/scripting/api/libs/graphics.cpp

```diff
@@ -502,31 +519,56 @@ ADE_FUNC(setCamera, l_Graphics, "[camera Camera]", "Sets current camera, or rese
 	return ADE_RETURN_TRUE;
 }
 
-ADE_FUNC(setColor, l_Graphics, "number Red, number Green, number Blue, [number Alpha]",
-         "Sets 2D drawing color; each color number should be from 0 (darkest) to 255 (brightest)", nullptr, nullptr)
+ADE_FUNC(setColor,
+	l_Graphics,
+	"number Red, number Green, number Blue, [number Alpha] | color Color",
+	"Sets 2D drawing color; each color number should be from 0 (darkest) to 255 (brightest)",
+	nullptr,
+	nullptr)
 {
-	if(!Gr_inited)
+	if (!Gr_inited)
 		return ADE_RETURN_NIL;
 
-	int r,g,b,a=255;
+	int r, g, b, a = 255;
+	color col;
 
-	if(!ade_get_args(L, "iii|i", &r, &g, &b, &a))
-		return ADE_RETURN_NIL;
+	if (lua_isnumber(L, 1)) {
+		if (!ade_get_args(L, "iii|i", &r, &g, &b, &a))
+			return ADE_RETURN_NIL;
+
+		gr_init_alphacolor(&col, r, g, b, a);
 
-	color ac;
-	gr_init_alphacolor(&ac,r,g,b,a);
-	gr_set_color_fast(&ac);
+	} else {
+		gr_init_alphacolor(&col, 0, 0, 0, 255);
+
+		ade_get_args(L, "o", l_Color.Get(&col));
+	}
+
+	gr_set_color_fast(&col);
 
 	return ADE_RETURN_NIL;
 }
 
-ADE_FUNC(getColor, l_Graphics, nullptr, "Gets the active 2D drawing color", "number, number, number, number" , "rgba color which is currently in use for 2D drawing")
+ADE_FUNC(getColor,
+	l_Graphics,
+	"boolean",
+	"Gets the active 2D drawing color. True to return raw rgb, false to return a color object. Defaults to true.",
```

Comment:
I'd prefer true to return a color object, and false to return raw rgb, defaulting to false -- with the rationale that setting a flag to true changes the default behavior -- but I won't make that a condition of the review.